### DISCUSSION TITLE
VCPKG_DISABLE_COMPILER_TRACKING when switching compiler

### DISCUSSION THREAD

**Discussion by niyue** [microsoft/vcpkg#22621] [Q&A] [answered]
I use vcpkg manifest mode to specify my project dependencies, and all these dependencies are pre-built using gcc under Linux and stored in `$VCPKG_ROOT/archives` directory. 

I would like to build my application using `clang` instead of gcc (gcc is too memory consuming in my project when building), so I specify the `CXX` environment variable to switch compiler from `gcc` to `clang`, and I specify environment variable `VCPKG_DISABLE_COMPILER_TRACKING=true` from the very beginning (https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_disable_compiler_tracking), but vcpkg still tries to re-built every dependency again when running `vcpkg install` to build my application.

I would like to know if this is an expected behavior (it seems to me `VCPKG_DISABLE_COMPILER_TRACKING` is not working), and if it is expected behavior, is there any approach 1) I can build the dependencies using `gcc` 2) while building the application using these dependencies with another compiler (`clang`) without re-building all these dependencies? Thanks.

**Comment by autoantwort** [comment, marked as answer]
> https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_disable_compiler_tracking

When I understand the documentation correctly, you have to set the variable inside of an own triplet.

**Reply by niyue** [reply]
Thanks so much. There are "variables" with similar names like `VCPKG_DEFAULT_TRIPLET` that is set via environment variable, and I got confused where they should be set.
